### DISCUSSION TITLE
Vertex weights for get_laplacian

### DISCUSSION THREAD

**Discussion by msseibel** [pyg-team/pytorch_geometric#2523] [Ideas]
The current implementation of get_laplacian provides 3 types of normalization for the Laplacian (`None`, `sym` , `rw`).

For the following discussion I refer to Bronstein et al. "Geometric deep learning: going beyond Euclidean data" .

The `None` and `rw` normalization techniques can be derived from L = A⁻¹ (D - W). 
D is the degree matrix: D = np.diag(np.sum(W,axis=1)) - np.diagonal(W).
A is the matrix of vertex weights.

For obtaining the unnormalized Laplacian (`None`) we set A = Identity. 
To get `rw`we set A = D.

Can we have a more general interface where we specify the matrix A?

**Comment by rusty1s** [comment]
We currently do not support the notion of defining vertex weights in `get_laplacian`, but I'm definitely open for a more general interface. Do you have a use-case where this might be important?

**Reply by msseibel** [reply]
Thank you for the fast reply.

I have an [approach](https://spharapy.readthedocs.io/en/latest/introduction.html) from traditional signal processing where the Laplacian for a manifold is defined as above. 
In short: A signal (electric potential) **f** is sampled at vertices of a triangular mesh. The authors use the finite element method (FEM) and assume that **f** changes linearly from one vertex to another. The matrix **A** is a result from the FEM discretization. I can imagine that the FEM discretization is also useful for modelling other physical cases (e.g. heat transfer).

I wanted to show that geometric deep learning generalizes this approach. 

Luckily the authors also use the cotangent weights for discretization where I don't need the matrix **A**. So I will stick with the simpler approach for now.
